pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -101,12 +101,33 @@ def teardown(self) -> None:
             signal.signal(signum, handler)
         self._original_handlers = {}
 
-    def _get_current_signal_handlers(self) -> _SIGNAL_HANDLER_DICT:
-        """Collects the currently assigned signal handlers that are relevant for Lightning."""
-        handlers = {signal.SIGTERM: signal.getsignal(signal.SIGTERM)}
-        if not self._is_on_windows():
-            handlers[signal.SIGUSR1] = signal.getsignal(signal.SIGUSR1)
-        return handlers
+    @staticmethod
+    def _get_current_signal_handlers() -> _SIGNAL_HANDLER_DICT:
+        """Collects the currently assigned signal handlers."""
+        valid_signals = SignalConnector._valid_signals()
+        if not _IS_WINDOWS:
+            # SIGKILL and SIGSTOP are not allowed to be modified by the user
+            valid_signals -= {signal.SIGKILL, signal.SIGSTOP}
+        return {signum: signal.getsignal(signum) for signum in valid_signals}
+
+    @staticmethod
+    def _valid_signals() -> Set[Signals]:
+        """Returns all valid signals supported on the current platform. Behaves identically to
+        :func:`signals.valid_signals` in Python 3.8+ and implements the equivalent behavior for older Python
+        versions."""
+        if sys.version_info >= (3, 8):
+            return signal.valid_signals()
+        elif _IS_WINDOWS:
+            return {
+                signal.SIGABRT,
+                signal.SIGFPE,
+                signal.SIGILL,
+                signal.SIGINT,
+                signal.SIGSEGV,
+                signal.SIGTERM,
+                signal.SIGBREAK,
```

Comment:
How did you find which signals go into this list? adding a comment with the reference would be nice.

Reply:
from the Python docs :) added reference